spanner/value.go

```diff
@@ -5592,12 +5592,26 @@ func encodeProtoEnumArray(len int, at func(int) reflect.Value) (*proto3.Value, e
 	return listProto(vs...), nil
 }
 
+// spannerTag contains metadata about a struct field's spanner tag.
+type spannerTag struct {
+	// Generated is true if the field is a generated column.
+	Generated bool
+}
+
 func spannerTagParser(t reflect.StructTag) (name string, keep bool, other interface{}, err error) {
 	if s := t.Get("spanner"); s != "" {
 		if s == "-" {
 			return "", false, nil, nil
 		}
-		return s, true, nil, nil
+		parts := strings.Split(s, ",")
+		name = parts[0]
+		tag := spannerTag{}
+		for _, part := range parts[1:] {
+			if part == "generated" {
```

Comment:
As mentioned above; Should we consider using `readOnly` instead for this? Or is there prior art / precedence from other databases or frameworks that indicate that `generated` is a more common term?

Reply:
Updated PR to use `readOnly` as well as how GORM works parses this `->`